src/audio/pipeline.c

```diff
@@ -824,12 +824,15 @@ static int pipeline_comp_reset(struct comp_dev *current,
 	struct pipeline *p = ctx->comp_data;
 	int stream_direction = dir;
 	int end_type;
+	int is_single_ppl = comp_is_single_pipeline(current, p->source_comp);
+	int is_same_sched =
+		pipeline_is_same_sched_comp(current->pipeline, p);
 	int err;
 
 	pipe_dbg(current->pipeline, "pipeline_comp_reset(), current->comp.id = %u, dir = %u",
 		 dev_comp_id(current), dir);
 
-	if (!comp_is_single_pipeline(current, p->source_comp)) {
+	if (!is_single_ppl && !is_same_sched) {
```

Comment:
Can you add an inline comment here for this conditional since the use case is complex. I'm assuming you are checking for two connected pipelines taht are scheduled with different components ?

Reply:
@lgirdwood 
I've updated comment. I'm checking pipelines that are scheduled with the same schedule component. ATM it is the sam as in ```pipeline_comp_trigger()```:
``` c
/* trigger should propagate to the connected pipelines,
	 * which need to be scheduled together
	 */
	if (!is_single_ppl && !is_same_sched) {
		pipe_dbg(current->pipeline, "pipeline_comp_trigger(), current is from another pipeline");
```